src/rapids_pre_commit_hooks/lint.py

```diff
@@ -0,0 +1,222 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import bisect
+import contextlib
+import functools
+from functools import reduce
+
+
+class OverlappingReplacementsError(RuntimeError):
+    pass
+
+
+class Replacement:
+    def __init__(self, pos, newtext):
+        self.pos = pos
+        self.newtext = newtext
+
+    def __eq__(self, other):
+        if not isinstance(other, Replacement):
+            return False
+        return self.pos == other.pos and self.newtext == other.newtext
+
+    def __repr__(self):
+        return f"Replacement(pos={self.pos}, newtext={self.newtext})"
+
+
+class Warning:
```

Comment:
Let's avoid shadowing the builtin class `Warning`. Can we name this something like `ReplacementWarning`?

Reply:
Renamed to `LintWarning`.